WeaponPack/src/main/java/module-info.java

```diff
@@ -1,2 +1,17 @@
+import com.github.sef24sp4.common.projectile.ProjectileSPI;
+import com.github.sef24sp4.common.services.IEntityProcessingService;
+import com.github.sef24sp4.common.services.IGamePluginService;
+import com.github.sef24sp4.common.weapon.WeaponSPI;
+import com.github.sef24sp4.weaponpack.machinegun.MachineGunBulletControlSystem;
+import com.github.sef24sp4.weaponpack.machinegun.MachineGun;
+import com.github.sef24sp4.weaponpack.BulletPlugin;
+
 module WeaponPack {
+	requires Common;
+	requires CommonProjectile;
+	requires CommonWeapon;
+	provides IGamePluginService with BulletPlugin;
+	provides WeaponSPI with MachineGun;
+	provides ProjectileSPI with MachineGunBulletControlSystem;
```

Comment:
Remove these provides. And instead provide a [`WeaponItem`](https://github.com/SE-F24-SP4/semesterproject-4/blob/040aeac44407afebf1826d8681cd832073758358/CommonItem/src/main/java/com/github/sef24sp4/common/item/itemtypes/WeaponItem.java#L6) for MachineGun and ShotGun.